src/Settings/settings.json

```diff
@@ -429,7 +447,25 @@
         "name": "Replace Tabs",
         "desc": "Replace tabs by spaces",
         "type": "bool",
-        "tip": "Use spaces instead of a tab character."
+        "tip": "Use spaces instead of a tab character.",
+        "depends": [
+            {
+                "name": "FakeVim/Enable",
+                "check": "return !var.toBool();"
+            }
+        ]
+    },
+    {
+        "name": "Cursor Overwrite",
+        "desc": "Default cursor overwrites",
+        "type": "bool",
+        "tip": "Default cursor is the cursor you see on opening new tabs.\nIf enabled, default cursor will overwrite next character.\nThis behaviour can be toggled for each tab by pressing INS/Insert on your keyboard.",
+        "depends": [
+            {
+                "name": "FakeVim/Enable",
+                "check": "return !var.toBool();"
+            }
+        ]
```

Comment:
This is something not related with fakevim.

Cursor in normal editing is of two forms, INSERT and REPLACE. In insert mode cursor is between characters and new characters is inserted in between.

In REPLACE mode, cursor is always over a character and new character overwrites the character over which cursor is present.

This setting tells when new tabs are opened in which mode the cursor should be in non-vim mode? By default it is in INSERT mode. For those who like to code in REPLACE mode will have to press `INS` to change cursor mode everytime new tabs are opened. This setting just helps them in it.

The behaviour of toggling edit mode was not present in QCodeEditor prior to this PR. So you can check the related QCodeEditor PR.

Reply:
> For those who like to code in REPLACE mode will have to press `INS` to change cursor mode everytime new tabs are opened. This setting just helps them in it.

I'm wondering if this is helpful to anyone :thinking: Do you know anyone who likes to code in replace mode?

Anyhow, the name "Default cursor overwrites" really confused me.

Reply:
In vim mode there is a way to make cursor be in replace or insert mode. In all popular code editor you have ability to toggle Insert and Replace mode. 

As for the name, Do you think "Use replace mode for default cursor" is less confusing? If you want we can completely get rid of this setting as well. User will have to manually toggle it by pressing INS, the setting is nothing more than a convienience.

As for who uses it, I am sure people who code in old environment use it, like tourist.

Reply:
> In all popular code editor you have ability to toggle Insert and Replace mode.

Do you mean toggle during coding or toggle the default mode? Can you give an emample of a setting to toggle the default mode?

Reply:
> Do you mean toggle during coding or toggle the default mode? 

Toggling during coding.

> Can you give an emample of a setting to toggle the default mode?

I haven't found any because I haven't used Replace mode ever and bothered to look for such settings.

I will remove this setting.